Sources/TuistCache/GraphMappers/CacheTreeShakingGraphMapper.swift

```diff
@@ -6,9 +6,52 @@ import TuistGraph
 public final class CacheTreeShakingGraphMapper: GraphMapping {
     public init() {}
 
+    public func map(graph: ValueGraph) throws -> (ValueGraph, [SideEffectDescriptor]) {
```

Comment:
As a suggestion for the `ValueGraph` perhaps a simpler approach to "tree shaking" is to reconstruct the graph from the entry point(s) using the `ValueGraphLoader` - this would ensure any targets / projects that aren't referenced are not included in the final graph.

https://github.com/tuist/tuist/blob/main/Tests/TuistCoreTests/ValueGraph/ValueGraphLoaderTests.swift#L27

Would that help at all?

Reply:
That's in fact a good idea. However, I'd do it as part of a different PR. I'd like to focus on replicating the logic as it is, to remove `Graph` soon, and then we can revisit this whole thing. I have the feeling we can get rid of this all together.